### PR TITLE
#885 disables move tab to a new window when only one tab is opened

### LINKED ISSUE #885: A container tab can wrongly be moved to a new window even if it's the only opened tab
**[Affected versions]:**
- Multi-Account Containers 4.0.3
- Firefox 56 and up

**[Affected Platforms]:**
- All Windows
- All Mac
- All Linux

**[Prerequisites]:**
- Have a Firefox profile with the latest "Multi-Account Containers" add-on version (4.0.3) installed.

**[Steps to reproduce]:**
1. Open the browser with the profiles from prerequisites.
2. Open a website in the "Personal" container and close other tabs if they exist.
3. Click on the "Multi-Account Containers" button from the toolbar.
4. Click on "View Personal container" button (>) and observe the "Move tabs to a new window" button.

**[Expected result]:**
- The button is greyed out.

**[Actual result]:**
- The button is active.

**[Notes]:**
- In our opinion this option should be greyed out because its effect uses system resources and does not do anything.
- In the same context, "Move to a new window" option from Firefox context menu is greyed out.
- The issue is reproducible with all containers.
- The issue is also reproducible with the TxP version of "Multi-Account Containers".
- Attached a screen recording of the issue:
new window

### PR DESCRIPTION
**Pull request by ShivangiKakkar**
### Description:
- Resolves the issue of container tab can wrongly be moved to a new window even if it's the only opened tab.

### CODE REVIEW

**Comment by ShivangiKakkar** [comment]
@groovecoder Thanks for assigning me this issue. This is my first contribution here, looking forward to contribute a lot more. :)
Please review.

File: src/js/popup.js [hunk 1]
```diff
@@ -212,6 +212,11 @@ const Logic = {
     return false;
   },
 
+  async noOfTabs() {
```
Comment (groovecoder) [inline src/js/popup.js]: nitpick: can the function name be `numOfTabs`?

File: src/js/popup.js [hunk 2]
```diff
@@ -717,13 +722,21 @@ Logic.registerPanel(P_CONTAINER_INFO, {
...
         moveTabsEl.parentNode.insertBefore(fragment, moveTabsEl.nextSibling);
       } else {
-        Logic.addEnterHandler(moveTabsEl, async function () {
-          await browser.runtime.sendMessage({
-            method: "moveTabsToWindow",
-            windowId: browser.windows.WINDOW_ID_CURRENT,
-            cookieStoreId: Logic.currentIdentity().cookieStoreId,
-          });
-          window.close();
+        Logic.noOfTabs().then(result => {
```
Comment (groovecoder) [inline src/js/popup.js]: Could you change this to `const numTabs = await Logic.numTabs()` so we don't have to use `then(...)`?

File: src/js/popup.js (hunk 1 above)
Comment (ShivangiKakkar) [reply src/js/popup.js]: sure :)

Comment (ShivangiKakkar) [reply src/js/popup.js]: > nitpick: can the function name be `numOfTabs`?

Sure :)

File: src/js/popup.js (hunk 2 above)
Comment (ShivangiKakkar) [reply src/js/popup.js]: > Could you change this to `const numTabs = await Logic.numTabs()` so we don't have to use `then(...)`?

changing it right away!

**Comment by ShivangiKakkar** [comment]
@groovecoder Review changes are done. Please take a look. Thanks :)

**Comment by ShivangiKakkar** [comment]
@groovecoder Yes, this looks much more cleaner. Changing it. Thanks :)

File: src/js/popup.js [hunk 3]
```diff
@@ -717,14 +722,23 @@ Logic.registerPanel(P_CONTAINER_INFO, {
...
+                method: "moveTabsToWindow",
+                windowId: browser.windows.WINDOW_ID_CURRENT,
+                cookieStoreId: Logic.currentIdentity().cookieStoreId,
+              });
+              window.close();
+            });
+          }
+        } catch (e) {
+          throw new Error("Could not get the number of active tabs.");
+        }
```
[the remark this answers is not in the document]
Comment (ShivangiKakkar) [reply src/js/popup.js]: Yes, this looks much more cleaner. Changing it. Thanks :)

**Comment by groovecoder** [comment]
Looks good. Thanks! I'm merging now, but I'm not sure when we'll do another release.

**Comment by ShivangiKakkar** [comment]
@groovecoder Thanks for guiding throughout. :)

**Pull request merged**